services/collectd/service.go

```diff
@@ -53,13 +52,28 @@ func TypesDBFile(path string) (typesdb *api.TypesDB, err error) {
 	return
 }
 
+type Diagnostic interface {
+	Starting()
+	Listening(addr net.Addr)
+	Closed()
+	UnableToReadDirectory(path string, err error)
+	LoadingPath(path string)
+	TypesParseError(path string, err error)
+
+	ReadFromUDPError(err error)
+	ParseError(err error)
+	DroppingPoint(name string, err error)
+	InternalStorageCreateError(err error)
```

Comment:
I really like how having specific errors like this. I know I said it previously, just felt like saying it again.